[API Node] Add cost indicators on API nodes

christian-byrne opened the pull request:
Adds text to API nodes' badges indicating their cost per run (estimation). For reference: https://docs.comfy.org/tutorials/api-nodes/overview#pricing

┆Issue is synchronized with this [Notion page](https://www.notion.so/PR-3924-API-Node-Add-cost-indicators-on-API-nodes-1f76d73d3650814aa80ae1f117f66425) by [Unito](https://www.unito.io)

christian-byrne committed:
feat: add API node pricing display in node badges

- Adds pricing data for API nodes from various providers
- Displays pricing in the credits badge next to API nodes
- Implements pricing display in gold badge color to distinguish costs

christian-byrne committed:
refactor: improve API node pricing display
- Remove inaccurate fallback price for unknown nodes
- Add user setting to toggle price badges
- Show badge only when pricing info is available
- Consolidate pricing data to single source of truth

christian-byrne committed:
fix: always show API badge icon regardless of price availability

christian-byrne committed:
chore: remove unused apiNodeNames.ts file

christian-byrne committed:
chore: improve API node pricing settings and comments
- Move API node pricing setting to 'Comfy > API Nodes' category
- Remove 'GENERATED FILE' header comment from apiNodeCosts.ts

Commit:
Update locales [skip ci]

christian-byrne committed:
fix: use node.constructor.nodeData.name instead of node.constructor.name for API node cost lookup

Fixed API node cost indicators by using the correct node name property for price lookup. The previous implementation incorrectly used the constructor name ('ComfyNode') instead of the actual node type name from nodeData.name.

huchenlei commented on src/types/apiNodeTypes.ts:
```diff
@@ -2,3 +2,14 @@ export interface ApiNodeCost {
   cost: number
 }
+
+export interface ApiNodeCostData {
```
Shall we document each field here?

nit: Do we really need `vendor` field if in the real data, the field is always `Unknown`?

huchenlei commented on src/types/apiNodeTypes.ts:
```diff
@@ -2,3 +2,14 @@ export interface ApiNodeCost {
+  pricingParams: string
+  pricePerRunRange: string
+  displayPrice: string
+  rateDocumentation?: string
```
nit: we should indicate that this field is a URL

christian-byrne committed:
fix: update API node costs with correct vendor information and pricing data from CSV

christian-byrne committed:
fix: add documentation to API node types and rename rateDocumentation to rateDocumentationUrl

christian-byrne commented:
Let me also add Luma as `Variable (by pixel)`

christian-byrne committed:
feat: add missing Luma nodes and indicate high pricing for Google Veo2

christian-byrne committed:
feat: improve API node pricing display with low/medium/high classifications

christian-byrne committed:
fix: restore correct /bin/bash.004 price for Recraft Crisp Upscale Image

webfiltered reviewed (changes requested):
Just cleanup nits (assume from refactors).

christian-byrne committed:
refactor

christian-byrne commented on src/composables/node/useNodePricing.ts:
```diff
@@ -0,0 +1,33 @@
+    return ''
+  }
+  return getNodeDisplayPrice(node.constructor.nodeData.name)
+}
```
Thanks, not sure what is going on with Claude there.

christian-byrne committed:
[refactor] Move static apiNodeCosts data outside composable scope

As suggested in PR review, moved the static pricing data structure to module level
to avoid recreating it on every useNodePricing() call, improving performance.

christian-byrne committed:
remove useless AI-generated comment

christian-byrne committed:
Merge branch 'main' into api-node-cost-indicator-v2

Commit:
Update locales [skip ci]

huchenlei committed:
Merge branch 'main' into api-node-cost-indicator-v2
